### ISSUE TITLE
Markdown: textual numbers are mistaken for `<pre>` characters at margins

### ISSUE THREAD

**Issue by rdipardo** [ScintillaOrg/lexilla#117]
At the start of every line, `LexMarkdown::Lex` defaults to `SCE_MARKDOWN_PRECHAR` in the absence of any reserved markup token:

https://github.com/ScintillaOrg/lexilla/blob/31a581550593eef5d0927d1f1bea1df14beb43c6/lexers/LexMarkdown.cxx#L341-L344

This is supposed to set off indented code blocks by filling the leading white space. But a textual number with no leading white space is also filled as if the digits were `<pre>` characters:

*Text:*
~~~markdown
The number:

338269006135764734700913562171

is prime
~~~

*Styles:*
~~~
{0}The number:{1}

{12}338269006135764734700913562171{1}

{0}is prime{1}
~~~

![scite-531-mkd](https://user-images.githubusercontent.com/59004801/201459051-38346274-dc5c-4911-8843-854ff40eccbf.png)

Every digit is captured because a subsequent code path increments the style region as if the number were an ordered list item:

https://github.com/ScintillaOrg/lexilla/blob/31a581550593eef5d0927d1f1bea1df14beb43c6/lexers/LexMarkdown.cxx#L380-L384

This patch adds a corrective fallback to `SCE_MARKDOWN_DEFAULT` as soon as a string of digits fails to match a list item:

[0001-LexMarkdown-Fix-numbers-as-pre-char.diff.txt](https://github.com/ScintillaOrg/lexilla/files/9994226/0001-LexMarkdown-Fix-numbers-as-pre-char.diff.txt)

**Comment by rdipardo** [comment]
These are the tests: [Issue117-Tests.diff.txt](https://github.com/ScintillaOrg/lexilla/files/9994247/Issue117-Tests.diff.txt)

**Comment by nyamatongwe** [comment]
Included in 5.2.1 release.
